src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,1984 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Represents a shortened (16-bit) version of 32 bit floating-point value (<see cref="float"/>).
+    /// </summary>
+    public readonly struct BFloat16
+        : IComparable,
+          ISpanFormattable,
+          IComparable<BFloat16>,
+          IEquatable<BFloat16>,
+          IBinaryFloatingPointIeee754<BFloat16>,
+          IMinMaxValue<BFloat16>,
+          IUtf8SpanFormattable,
+          IBinaryFloatParseAndFormatInfo<BFloat16>
+    {
+        private const NumberStyles DefaultParseStyle = NumberStyles.Float | NumberStyles.AllowThousands;
+
+        // Constants for manipulating the private bit-representation
+
+        internal const ushort SignMask = 0x8000;
+        internal const int SignShift = 15;
+        internal const byte ShiftedSignMask = SignMask >> SignShift;
+
+        internal const ushort BiasedExponentMask = 0x7F80;
+        internal const int BiasedExponentShift = 7;
+        internal const int BiasedExponentLength = 8;
+        internal const byte ShiftedBiasedExponentMask = BiasedExponentMask >> BiasedExponentShift;
+
+        internal const ushort TrailingSignificandMask = 0x007F;
+
+        internal const byte MinSign = 0;
+        internal const byte MaxSign = 1;
+
+        internal const byte MinBiasedExponent = 0x00;
+        internal const byte MaxBiasedExponent = 0xFF;
+
+        internal const byte ExponentBias = 127;
+
+        internal const sbyte MinExponent = -126;
+        internal const sbyte MaxExponent = +127;
+
+        internal const ushort MinTrailingSignificand = 0x0000;
+        internal const ushort MaxTrailingSignificand = 0x007F;
+
+        internal const int TrailingSignificandLength = 7;
+        internal const int SignificandLength = TrailingSignificandLength + 1;
+
+        // Constants representing the private bit-representation for various default values
+
+        private const ushort PositiveZeroBits = 0x0000;
+        private const ushort NegativeZeroBits = 0x8000;
+
+        private const ushort EpsilonBits = 0x0001;
+
+        private const ushort PositiveInfinityBits = 0x7F80;
+        private const ushort NegativeInfinityBits = 0xFF80;
+
+        private const ushort PositiveQNaNBits = 0x7FC0;
```

Comment:
You're getting a warning, so probably best to remove this or just comment it out

Reply:
Half.PositiveQNaNBits also seems unused, but not get warned. Curious here.